File: C++/A/738A-interviewwitholeg.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    string s;
    cin >> s;
    stringstream interviewText;
    for (int i = 0; i < n; i++)
    {
        if (s[i] == 'o' && i + 2 < n && s[i + 1] == 'g' && s[i + 2] == 'o')
        {
            interviewText << "***";
            i += 2;
            while (i + 1 < n && s[i + 1] == 'g' && s[i + 2] == 'o')
                i += 2;
        }
        else
            interviewText << s[i];
    }

    cout << interviewText.str();
}

File: Python/A/738A-interviewwitholeg.py
n = int(input())
s = input()
interview_text = ""
i = 0

while i < n:
    if s[i:i+3] == 'ogo':
        interview_text += "***"
        i += 2
        while i + 1 < n and s[i + 1:i + 3] == 'go':
            i += 2
    else:
        interview_text += s[i]
    i += 1

print(interview_text)
